src/react-apps/applications/runtime/src/features/form/rules/sagas/rule/index.ts

```diff
@@ -1,33 +1,138 @@
 import { SagaIterator } from 'redux-saga';
-import { call, takeLatest } from 'redux-saga/effects';
-
+import { call, select, takeLatest } from 'redux-saga/effects';
+import * as RuleActions from '../../actions/rule';
 import * as ActionTypes from '../../actions/types';
-import { ICheckIfRuleShouldRun } from '../../actions/rule';
+
+import { IFormDynamicState } from '../../../dynamics/reducer';
+import { IFormRuleState } from '../../reducer';
+import { IFormData } from '../../../data/reducer';
+import FormDataActions from '../../../data/actions';
+import { IDataModelFieldElement } from '../../';
+import { ILayoutState } from '../../../layout/reducer';
+import { IDataModelState } from '../../../datamodell/reducer';
+import { IRuntimeState } from '../../../../../types';
+
+const selectRuleConnection = (state: IRuntimeState): IFormDynamicState => state.formDynamics.ruleConnection;
+const selectFormDataConnection = (state: IRuntimeState): IFormData => state.formData;
+const selectFormLayoutConnection = (state: IRuntimeState): ILayoutState => state.formLayout;
+const selectFormdataModelConnection = (state: IRuntimeState): IDataModelState => state.formDataModel;
 
 function* checkIfRuleShouldRunSaga({
   lastUpdatedComponentId,
-  lastUpdatedDataField,
-  lastUpdatedData,
+  lastUpdatedDataBinding,
+  lastUpdatedDataValue,
   repeatingContainerId,
-}: ICheckIfRuleShouldRun): SagaIterator {
+}: RuleActions.ICheckIfRuleShouldRun): SagaIterator {
   try {
-    yield call(
-      console.log,
-      'Check if Rule should run',
-      lastUpdatedComponentId,
-      lastUpdatedDataField,
-      lastUpdatedData,
-      repeatingContainerId,
-    )
+    const ruleConnectionState: IFormRuleState = yield select(selectRuleConnection);
+    const formDataState: IFormData = yield select(selectFormDataConnection);
+    const formLayoutState: ILayoutState = yield select(selectFormLayoutConnection);
+    const formDataModelState: IDataModelState = yield select(selectFormdataModelConnection);
+    let repContainer;
+    let repeating;
+    let dataModelGroup: string;
+    let index;
+    if (repeatingContainerId) {
+      repContainer = formLayoutState.layout[repeatingContainerId];
+      repeating = repContainer.repeating;
+      dataModelGroup = repContainer.dataModelGroup;
+      index = repContainer.index;
+    }
+
+    const isPartOfRepeatingGroup: boolean = (repeating && dataModelGroup != null && index != null);
```

Comment:
I prefer moving logic out of our saga, as we have agreed, and then write tests for the logic.